src/cdk/a11y/_index.scss

```diff
@@ -54,12 +54,39 @@
   }
 }
 
+/// @deprecated Use `active` target instead.
+/// @breaking-change 18.0.0
+///
+/// Emits the mixin's content nested under a `.cdk-high-contrast-black-on-white` or
+/// `.cdk-high-contrast-white-on-black-on-white` selector if $target is black-on-white or
+/// white-on-black.
+///
+/// Supports the deprecated `black-on-white` and `white-on-black` targets, which depend on
+/// deprecated `HighContrastModeDetector`. To be removed when `HighContrastModeDetector` is
+/// removed.
+///
+/// @param {String} target Type of high contrast setting to target. Can be `active`,
+///     `white-on-black` or `black-on-white`.
+@mixin _optinally-nest-target-class($target) {
```

Comment:
This wasn't what I was referring to yesterday. I think that the `high-contrast` mixin implementation should change to this:

```scss
@mixin high-contrast($target: active, $encapsulation: 'any') {
  @if ($target != 'active' and $target != 'black-on-white' and $target != 'white-on-black') {
    @error 'Unknown cdk-high-contrast value "#{$target}" provided. ' +
           'Allowed values are "active", "black-on-white", and "white-on-black"';
  }

  @if ($encapsulation != 'on' and $encapsulation != 'off' and $encapsulation != 'any') {
    @error 'Unknown cdk-high-contrast encapsulation "#{$encapsulation}" provided. ' +
           'Allowed values are "on", "off", and "any"';
  }

  @if ($target == 'black-on-white' or $target == 'white-on-black') {
    @warn '...';
  }

  $value: if($target != 'active', 'none', 'active');

  @media(forced-colors: $value) {
    @content;
  }
}
```

Reply:
ohhh, ok. that looks fine to me.

Reply:
actually, that doesn't work because the legacy checkbox still needs to use `black-on-white` detection.

Reply:
Did you want legacy checkbox to use the `.cdk-high-contrast-black-on-white` selector instead of the mixin?